napari/components/_tests/test_layer_slicer.py

```diff
@@ -289,3 +292,30 @@ def test_slice_layers_async_with_one_3d_image(layer_slicer):
 
     layer_result = future.result()[layer]
     np.testing.assert_equal(layer_result.data, data[2, :, :])
+
+
+def test_slice_layers_async_with_one_3d_points(layer_slicer):
+    np.random.seed(0)
+    num_points = 100
+    data = np.rint(2.0 * np.random.rand(num_points, 3))
+    layer = Points(data=data)
+    lockable_internal_data = LockableData(layer._data)
```

Comment:
`Points.data` does not support the layer data protocol (I think only `Image` truly does). The initializer materializes `data` to a numpy array almost immediately: https://github.com/napari/napari/blob/f2a250833a4277256bad281d87a3338d843906a4/napari/layers/points/points.py#L336

So I don't think `LockableData` does anything here and so the `assert not future.done` is not guaranteed to pass.

Reply:
Yes, that is why I did this after initialization.

Reply:
Note that I'm directly manipulating `_data` from `Points` not the input `data` (which was the case during your first review).

Reply:
Ah sorry, I see. Even though this hacks at the internals a bit artificially, I don't mind it too much since it gives the execution control we want/need. However, I strongly suggest adding an explanatory/implementation comment next to the assignment to `layer._data` to clarify that's what's happening and why we're doing it here.